sqlglot/expressions.py

```diff
@@ -7386,12 +7386,19 @@ def convert(value: t.Any, copy: bool = False) -> Expression:
     if isinstance(value, bytes):
         return HexString(this=value.hex())
     if isinstance(value, datetime.datetime):
-        datetime_literal = Literal.string(
-            (value if value.tzinfo else value.replace(tzinfo=datetime.timezone.utc)).isoformat(
-                sep=" "
-            )
-        )
-        return TimeStrToTime(this=datetime_literal)
+        if not value.tzinfo:
+            value = value.replace(tzinfo=datetime.timezone.utc)
```

Comment:
why are you always defaulting to utc? if there's a UTC, do we want it to have a time zone?

Reply:
that's what the original code did?

`(value if value.tzinfo else value.replace(tzinfo=datetime.timezone.utc))`

Reply:
yea, but now you're distinguishing between timezones and not, right? this didn't exist in the pr.evious code

Reply:
The previous code would force all naive datetimes to be treated as UTC. I think this is a reasonable default because in my opinion it leads to the least amount of surprises when doing things like additional casts to other types or running the values through other date functions.

Wouldnt anyone relying on the existing behaviour also be relying on this?

Reply:
I mean its still _kind_ of naive because this change uses the original logic. Any UTC dates are cast to `TIMESTAMP`, not `TIMESTAMPTZ`, so the timezone is only taken into account for non-UTC

Reply:
ok that's fine

Reply:
Actually, I had to remove this assumption because engines like BigQuery dont support a timezone part in the string literal (eg `+08:00`). It was generating:

```
select CAST('2023-01-01 00:00:00+00:00' AS DATETIME)
```

for BigQuery which is invalid, it throws ` Invalid datetime string "2023-01-01 00:00:00+00:00"`

So now:
 - naive datetime objects remain naive, these will be the `TIMESTAMP` type
 - anything with `tzinfo` set will be `TIMESTAMPTZ`, even if `tzinfo` is just UTC

This causes the above expression to be generated as:
```
CAST('2023-01-01 12:13:14+00:00' AS TIMESTAMP)
```

for BigQuery, which is correct

Reply:
cool